sunpy/util/decorators.py

```diff
@@ -264,3 +266,47 @@ def __call__(self, func):
         if self.kwargs:
             func.__doc__ = func.__doc__.format(**self.kwargs)
         return func
+
+
+def deprecate_positional_args(f):
+    """
+    Decorator for methods that issues warnings for positional arguments
+    Using the keyword-only argument syntax in pep 3102, arguments after the
+    * will issue a warning when passed as a positional argument.
+
+    Parameters
+    ----------
+    f : function
+        Function to check arguments on.
+
+    References
+    ----------
+    Taken from from `scikit-learn <https://github.com/scikit-learn/scikit-learn/blob/master/sklearn/utils/validation.py#L1271>`__.
+    Licensed under the BSD, see "licenses/SCIKIT-LEARN.rst".
+    """
+    sig = signature(f)
+    kwonly_args = []
+    all_args = []
+
+    for name, param in sig.parameters.items():
+        if param.kind == Parameter.POSITIONAL_OR_KEYWORD:
+            all_args.append(name)
+        elif param.kind == Parameter.KEYWORD_ONLY:
+            kwonly_args.append(name)
+
+    @wraps(f)
+    def inner_f(*args, **kwargs):
+        extra_args = len(args) - len(all_args)
+        if extra_args > 0:
+            # ignore first 'self' argument for instance methods
+            args_msg = ['{}={}'.format(name, arg)
+                        for name, arg in zip(kwonly_args[:extra_args],
+                                             args[-extra_args:])]
+            last_supported_version = ".".join(map(str, get_removal_version("2.0")))
```

Comment:
Er, the version number when the deprecation started is hard-coded as "2.0" into this decorator...

Reply:
```suggestion
            last_supported_version = ".".join(map(str, get_removal_version(base_removal_version)))
```
Would a variable like this, which is a parameter in the outer wrapper with default value = `"2.0"`, be sufficient?

Reply:
Well, don't call it `base_removal_version`.  You can call it `since`, just like the argument for `get_removal_version()`.

Also, don't give it any default.  In theory we want this decorator to be used in general, and a default value doesn't make any sense.